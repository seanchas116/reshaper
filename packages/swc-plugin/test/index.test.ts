import { describe, expect, it } from "vitest";
import swc from "@swc/core";
import fs from "fs/promises";
import path from "path";

describe("test", () => {
  it("test", async () => {
    const inputJs = await fs.readFile(
      path.resolve(__dirname, "fixture/input.js"),
      "utf-8"
    );

    const output = await swc.transform(inputJs, {
      // Some options cannot be specified in .swcrc
      filename: "fixture/input.js",
      sourceMaps: true,
      // Input files are treated as module by default.
      isModule: true,

      // All options below can be configured via .swcrc
      jsc: {
        parser: {
          syntax: "typescript",
          tsx: true,
        },
        transform: {},
        experimental: {
          plugins: [
            [
              path.resolve(
                __dirname,
                "../target/wasm32-wasi/release/swc_plugin.wasm"
              ),
              {},
            ],
          ],
        },
      },
    });
    expect(output.code).toMatchSnapshot();
  });
});
