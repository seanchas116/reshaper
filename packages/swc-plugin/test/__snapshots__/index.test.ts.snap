// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test > test 1`] = `
"export var x = /*#__PURE__*/ React.createElement("div", {
    "data-reshaper-loc": "fixture/input.js:1"
}, /*#__PURE__*/ React.createElement("h1", {
    "data-reshaper-loc": "fixture/input.js:3"
}, "Hello"));
"
`;
